compiler/rustc_ast/src/tokenstream.rs

```diff
@@ -57,7 +57,9 @@ impl TokenTree {
         match (self, other) {
             (TokenTree::Token(token, _), TokenTree::Token(token2, _)) => token.kind == token2.kind,
             (TokenTree::Delimited(.., delim, tts), TokenTree::Delimited(.., delim2, tts2)) => {
-                delim == delim2 && tts.eq_unspanned(tts2)
+                delim == delim2
+                    && tts.len() == tts2.len()
+                    && tts.iter().zip(tts2.iter()).all(|(a, b)| a.eq_unspanned(b))
```

Comment:
```suggestion
                    && tts.iter().zip(tts2).all(|(a, b)| a.eq_unspanned(b))
```
Methods like `zip` usually take `IntoIter` and not `Iter`.

Reply:
yeah, I tried it with a compiling error:

```console
error[E0277]: `&tokenstream::TokenStream` is not an iterator
  --> compiler/rustc_ast/src/tokenstream.rs:62:35
   |
62 |                     && tts.iter().zip(tts2).all(|(a, b)| a.eq_unspanned(b))
   |                                   ^^^ `&tokenstream::TokenStream` is not an iterator
   |
   = help: the trait `Iterator` is not implemented for `&tokenstream::TokenStream`
   = note: required for `&tokenstream::TokenStream` to implement `IntoIterator`

error[E0599]: the method `all` exists for struct `Zip<TokenStreamIter<'_>, &TokenStream>`, but its trait bounds were not satisfied
  --> compiler/rustc_ast/src/tokenstream.rs:62:45
   |
62 |                     && tts.iter().zip(tts2).all(|(a, b)| a.eq_unspanned(b))
   |                                             ^^^ method cannot be called on `Zip<TokenStreamIter<'_>, &TokenStream>` due to unsatisfied trait bounds
   |
  ::: /Users/yukang/rust/library/core/src/iter/adapters/zip.rs:15:1
   |
15 | pub struct Zip<A, B> {
   | -------------------- doesn't satisfy `_: Iterator`
   |
   = note: the following trait bounds were not satisfied:
           `&tokenstream::TokenStream: Iterator`
           which is required by `Zip<TokenStreamIter<'_>, &tokenstream::TokenStream>: Iterator`
```